community/cloud-foundation/templates/kms/tests/integration/kms.bats

```diff
@@ -0,0 +1,135 @@
+        create_config
+        # Create service accounts to test IAM bindings.
+        gcloud iam service-accounts create "${KEYRING_NAME}" \
+            --display-name "Test KMS Service Account"
+    fi
+
+    # Per-test setup steps.
+}
+
+function teardown() {
+    # Global teardown; executed once per test file.
+    if [[ "$BATS_TEST_NUMBER" -eq "${#BATS_TEST_NAMES[@]}" ]]; then
+        delete_config
+        rm -f "${RANDOM_FILE}"
+        # Delete service account after tests had been completed.
+        gcloud --quiet iam service-accounts delete "${SA_NAME}"
+    fi
+
+    # Per-test teardown steps.
+}
+
+########## TESTS ##########
+
+@test "Creating deployment ${DEPLOYMENT_NAME} from ${CONFIG}" {
+    gcloud deployment-manager deployments create "${DEPLOYMENT_NAME}" \
+        --config "${CONFIG}" --project "${CLOUD_FOUNDATION_PROJECT_ID}"
+    [[ "$status" -eq 0 ]]
+}
+
+@test "Verifying the KeyRing ${KEYRING_NAME} was created " {
+    run gcloud kms keyrings list --location ${REGION} \
+        --format="value(name.scope(keyRings))" \
+        --filter="${KEYRING_NAME}"
+    [[ "$status" -eq 0 ]]
+    [[ "$output" =~ "${KEYRING_NAME}" ]]
+}
+
+@test "KEY ${KEY_NAME} is created in KeyRing ${KEYRING_NAME} " {
+    run gcloud kms keys list --location ${REGION} --keyring="${KEYRING_NAME}" \
+        --format="value(name.scope(cryptoKeys))" \
+        --filter="${KEY_NAME}"
+    [[ "$status" -eq 0 ]]
+    [[ "$output" =~ "${KEY_NAME}" ]]
+}
+
+@test "CryptoKey's PURPOSE is set to ${KEY_PURPOSE} " {
+    run gcloud kms keys describe ${KEY_NAME} --location ${REGION} \
+        --keyring="${KEYRING_NAME}" \
+        --format="value(purpose)"
+    [[ "$status" -eq 0 ]]
+    [[ "$output" =~ "${KEY_PURPOSE}" ]]
+}
+
+@test "protectionLevel on key ${KEY_NAME} = SOFTWARE " {
+    run gcloud kms keys describe ${KEY_NAME} --location ${REGION} \
+        --keyring="${KEYRING_NAME}" \
+        --format="value(versionTemplate.protectionLevel)"
+    [[ "$status" -eq 0 ]]
+    [[ "$output" -eq "SOFTWARE" ]]
+}
+
+@test "Enc algorithm on key ${KEY_NAME} = GOOGLE_SYMMETRIC_ENCRYPTION" {
+    run gcloud kms keys describe ${KEY_NAME} --location ${REGION} \
+        --keyring="${KEYRING_NAME}" \
+        --format="value(versionTemplate.algorithm)"
+    [[ "$status" -eq 0 ]]
+    [[ "$output" -eq "GOOGLE_SYMMETRIC_ENCRYPTION" ]]
+}
+
+@test "Verify whether ${SA_NAME} has role ${ROLE} " {
+    run gcloud kms keys get-iam-policy ${KEY_NAME} --location ${REGION} \
+        --keyring="${KEYRING_NAME}" \
+        --format="value(bindings.role)"
+    [[ "$status" -eq 0 ]]
+    [[ "$output" =~  "${ROLE}" ]]
+}
+
+@test "Verify if ${SA_NAME} has access to ${KEY_NAME} " {
+    run gcloud kms keys get-iam-policy ${KEY_NAME} --location ${REGION} \
+        --keyring="${KEYRING_NAME}" \
+        --format="value(bindings.members[0])"
+    [[ "$status" -eq 0 ]]
+    [[ "$output" =~ "${SA_FQDN}" ]]
+}
+
+# Note: There is no Delete Deployment step because KeyRings and Keys 
```

Comment:
This needs a further discussion.

Reply:
Do you mean the `delete deployment` part? Based on this https://cloud.google.com/kms/docs/object-hierarchy#lifetime only key versions can be deleted.
In any case, if you need more clarification please let me know I can setup a meeting.